File: src/main/java/me/equixz/chatmood/mixin/OnChatMixin.java
package me.equixz.chatmood.mixin;

import me.equixz.chatmood.functions.ReceiveMessage;
import net.minecraft.client.network.ClientPlayNetworkHandler;
import net.minecraft.network.packet.s2c.play.GameMessageS2CPacket;
import net.minecraft.text.Text;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

@Mixin(ClientPlayNetworkHandler.class)
public class OnChatMixin {
    @Inject(method = "onGameMessage", at = @At("HEAD"))
    private void onGameMessage(GameMessageS2CPacket packet, CallbackInfo ci) {
        String message = packet.content().getString();
        ReceiveMessage.getReceiveMessageInstance().receiveMessages(Text.of(message));
    }
}

File: src/main/java/me/equixz/chatmood/mixin/onChatMixin.kt
package me.equixz.chatmood.mixin

import me.equixz.chatmood.functions.receiveMessage
import net.minecraft.client.network.ClientPlayNetworkHandler
import net.minecraft.network.packet.s2c.play.GameMessageS2CPacket
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo

@Mixin(ClientPlayNetworkHandler::class)
class onChatMixin {
    @Inject(
        method = ["onGameMessage(Lnet/minecraft/network/packet/s2c/play/GameMessageS2CPacket;)V"],
        at = [At("HEAD")]
    )
    private fun onGameMessage(packet: GameMessageS2CPacket, ci: CallbackInfo) {
        val message = packet.content()
        receiveMessage.getReceiveMessageInstance().receiveMessages(message)
    }
}
